stacks/resources/cognitoOidc/handlers/controller/linkedin.ts

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { got } from 'got';
+import { URLSearchParams } from 'url';
+import { ENV_COGNITO_REDIRECT_URI } from '../../types';
+import { makeIdToken } from '../crypto';
+import { filterOutScopesForLinkedin } from '../util';
+import * as req from './requests';
+
+// `openid` is an scope required by Cognito
+// but Linkedin throws an error if this scope is passed as it is not recognized
+// so it needs to be filtered out in the getAuthorizeUrl function
+export const linkedinScope = ['openid', 'r_liteprofile', 'r_emailaddress'];
+
+const linkedinApiUrl = 'https://api.linkedin.com';
+const linkedinLoginUrl = 'https://linkedin.com';
+
+interface UserDetails {
+  id: string;
+  firstName?: string;
+  lastName?: string;
+  localizedFirstName?: string;
+  localizedLastName?: string;
+  profilePicture?: any; // https://docs.microsoft.com/en-us/linkedin/shared/references/v2/profile/profile-picture
+  localizedHeadline?: string;
+  vanityName?: string;
+}
+
+const getApiEndpoints = (apiBaseUrl: string, loginBaseUrl: string) => {
+  return {
+    // https://docs.microsoft.com/en-us/linkedin/shared/references/v2/profile/lite-profile
+    userDetails: `${apiBaseUrl}/v2/me?projection=(id,vanityName,localizedFirstName,localizedLastName,profilePicture(displayImage~digitalmediaAsset:playableStreams),firstName,lastName)`,
```

Comment:
maybe also add the link to what projections are in linkedin?
`https://docs.microsoft.com/en-us/linkedin/shared/api-guide/concepts/projections?view=li-lms-2022-08`

Reply:
ok added